### ISSUE TITLE
"Failed address validation: Email has a wrong format"

### ISSUE THREAD

**Issue by Serializator** [magmodules/magento2-channable#106]
### Description
We've been debugging an issue since we configured the Channable module to not import customers ("Stores > Configuration > Magmodules > Order Import > Import Customers" => "No").

Since we configured the Channable module not to import customers we kept getting the below response when Channable tried to send orders to Magento (`/channable/order/hook`).

```json
{"validated":"false","errors":"Failed address validation:\nEmail has a wrong format"}
```

### Cause
Inside `Magento\Quote\Model\Quote\Address\ToOrderAddress::convert(Address $address, $data = [])` the `$data` argument is used to pass additional information into the resulting address (`$orderAddress`).

In `Magento\Quote\Model\QuoteManagement:submitQuote(QuoteEntity $quote, $orderData = [])` this method is used to convert the quote address into an order address (line 520 and 532).

In both cases the `$data` argument is passed with two keys. `address_type` and `email`, which are meant to additionally add "address_type" and "email" to the resulting order address. It though seems that `$quote->getCustomerEmail()` (which is used for the `email` key) is returning `null`, resulting in the e-mail being passed to be `null` as well (causing the validation failure).

**Why is this?** Depending on the checkout method for the quote (`$quote->getCheckoutMethod()`) the customer information is set (`Magento\Quote\Model\QuoteManagement::placeOrder` on line 387). But debugging this it seems that the checkout method is not set for the quote (`$quote->getCheckoutMethod()` returning `null`). Which is causing the customer e-mail not to be set etc.

### Possible Solution
Inside `Magmodules\Channable\Model\Order` the customer information of the quote is set depending on the `$this->importCustomer` ("Stores > Configuration > Magmodules > Order Import > Import Customers").

It might be enough to set the checkout method using `->setCheckoutMethod(CartManagementInterface::METHOD_GUEST)` method.

```php
 else {
    $customerId = 0;
    $cart->setCustomerId($customerId)
        ->setCustomerEmail($email)
        ->setCustomerFirstname($data['customer']['first_name'])
        ->setCustomerMiddlename($data['customer']['middle_name'])
        ->setCustomerLastname($data['customer']['last_name'])
        ->setCustomerIsGuest(true)
        ->setCustomerGroupId(GroupInterface::NOT_LOGGED_IN_ID)
        ** ->setCheckoutMethod(CartManagementInterface::METHOD_GUEST); **
}
```

**Comment by Marvin-Magmodules** [comment]
Hi @Serializator, thank you for the detailed issue report. I'm not able to reproduce this on CE 2.4.1 with the latest Channable version. What version of Magento did you test this on and did adding the setCheckoutMethod fixed the issue? 

Would like to reproduce this issue first.

**Comment by Serializator** [comment]
Hi @Marvin-Magmodules, we experienced this issue on Magento **EE 2.3.5-p1**, adding `setCheckoutMethod` fixed the problem for us. We're using version 1.4.21 of the Channable module.

**Comment by Marvin-Magmodules** [comment]
Hi @Serializator, unfortunately we were not able to reproduce this issue on 2.3.5-p1, but setting the checkout method to guest for guest orders makes sense. The suggested edit is merged and released; see [1.4.21](https://github.com/magmodules/magento2-channable/releases/tag/1.4.21) version. Thanks again for the detailed issue report. We are closing this issue now but please feel free to reopen the issue if this still occurs.
